craft_application/services/project.py

```diff
@@ -72,13 +72,17 @@ def configure(self, *, platform: str | None, build_for: str | None) -> None:
         if self.is_configured:
             raise RuntimeError("Project is already configured.")
 
-        platforms = self.get_platforms()
+        platforms = self.get_platforms().copy()
```

Comment:
We were accidentally modifying the raw platforms item!

Reply:
Good point! Also I realised this was still mutable by mutating the inner dicts so I fixed it properly.